cmd/clusterctl/client/repository/repository_github.go

```diff
@@ -388,19 +388,10 @@ func (g *gitHubRepository) downloadFilesFromRelease(release *github.RepositoryRe
 			return false, nil
 		}
 		if redirect != "" {
-			req, err := http.NewRequestWithContext(ctx, http.MethodGet, redirect, http.NoBody)
-			if err != nil {
-				retryError = errors.Wrapf(err, "failed to download file %q from %q release via redirect location %q: failed to create request", *release.TagName, fileName, redirect)
-				return false, nil
-			}
-
-			response, err := http.DefaultClient.Do(req) //nolint:bodyclose // (NB: The reader is actually closed in a defer)
-			if err != nil {
-				retryError = errors.Wrapf(err, "failed to download file %q from %q release via redirect location %q", *release.TagName, fileName, redirect)
-				return false, nil
-			}
-			reader = response.Body
+			retryError = errors.New("unexpected redirect: http.DefaultClient should be able to handle redirects in DownloadReleaseAsset")
```

Comment:
I believe this error eventually bubbles up to the error the user sees in the terminal output. It might be nice UX wise if the error read something like "unexpected redirect while downloading the release asset" (or something similar) instead of going into so much detail that might not be useful to the user.